### PR TITLE
fixed failing code, update go version, bump dependencies

### PR DESCRIPTION
**Pull request by stormeye2000**

### CODE REVIEW

File: tests/steps/speculative_execution_test.go [hunk 1]
```diff
@@ -60,10 +60,6 @@ func InitializeSpeculativeExecution(ctx *godog.ScenarioContext) {
 		})
 
 	ctx.Step(`^a valid speculative_exec_result will be returned with (\d+) transforms$`, func(transformCount int) error {
-		if len(speculativeExecResult.DeployHash.String()) == 0 {
```
Comment (meywood) [inline tests/steps/speculative_execution_test.go:63]: This should be changed to test correct data structure BlockHash

File: go.mod [hunk 2]
```diff
@@ -1,16 +1,17 @@
 module github.com/stormeye2000/cspr-sdk-standard-tests-go
 
-go 1.20
+go 1.21.6
 
 require (
 	github.com/acarl005/stripansi v0.0.0-20180116102854-5a71ef0e047d
 	github.com/cucumber/godog v0.12.6
-	github.com/make-software/casper-go-sdk v1.3.0
+	github.com/make-software/casper-go-sdk v1.4.5-0.20240126123809-b1f5443e71d2
```
Comment (meywood) [inline go.mod:8]: Do you need ".20240126123809-b1f5443e71d2"?

**Review by meywood — CHANGES_REQUESTED** [review]

File: tests/steps/speculative_execution_test.go (hunk 1 above)
Comment (stormeye2000) [reply tests/steps/speculative_execution_test.go:63]: done

File: go.mod (hunk 2 above)
Comment (stormeye2000) [reply go.mod:8]: yes

**Review by meywood — APPROVED** [review]

**Pull request merged**